vm/systemSmartContracts/delegation.go

```diff
@@ -2055,22 +2132,74 @@ func (d *delegation) getDelegatorFundsData(args *vmcommon.ContractCallInput) vmc
 	return vmcommon.Ok
 }
 
-func (d *delegation) getDelegatorList(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
-	returnCode := d.checkArgumentsForGeneralViewFunc(args)
+func (d *delegation) getTotalCumulatedRewardsForUser(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	delegator, returnCode := d.checkArgumentsForUserViewFunc(args)
 	if returnCode != vmcommon.Ok {
 		return returnCode
 	}
 
-	delegationStatus, err := d.getDelegationStatus()
+	err := d.computeAndUpdateRewards(args.Arguments[0], delegator)
 	if err != nil {
 		d.eei.AddReturnMessage(err.Error())
 		return vmcommon.UserError
 	}
 
-	for _, address := range delegationStatus.Delegators {
-		d.eei.Finish(address)
+	totalCumulatedRewards := big.NewInt(0).Add(delegator.TotalCumulatedRewards, delegator.UnClaimedRewards)
+	d.eei.Finish(totalCumulatedRewards.Bytes())
+
+	return vmcommon.Ok
+}
+
+func (d *delegation) setMetaData(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	returnCode := d.checkOwnerCallValueGasAndDuplicates(args)
+	if returnCode != vmcommon.Ok {
+		return returnCode
+	}
+
+	if len(args.Arguments) != 3 {
+		d.eei.AddReturnMessage("needed 3 arguments")
+		return vmcommon.UserError
 	}
 
+	dMetaData := &DelegationMetaData{
+		Name:        args.Arguments[0],
+		Description: args.Arguments[1],
+		URI:         args.Arguments[2],
+	}
+	marshaledData, err := d.marshalizer.Marshal(dMetaData)
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
+	d.eei.SetStorage([]byte(delegationMetaData), marshaledData)
+
+	return vmcommon.Ok
+}
+
+func (d *delegation) getMetaData(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	returnCode := d.checkArgumentsForGeneralViewFunc(args)
+	if returnCode != vmcommon.Ok {
+		return returnCode
+	}
+
+	marshaledData := d.eei.GetStorage([]byte(delegationMetaData))
+	if len(marshaledData) == 0 {
+		d.eei.AddReturnMessage("delegation meta data is not set")
```

Comment:
This can be easily tested

Reply:
would not test every if, as these are clear

Reply:
set and get meta data was already tested.

Reply:
It does not matter if the code is clear or not. tests should be there. Especially when we can easily write them. No excuses.